elastica/wrappers/base_system.py

```diff
@@ -131,6 +131,10 @@ def finalize(self):
 
         """
 
+        # FIXME: This is probably the most bizarre way to collect the functions. It is definitely
```

Comment:
Are there better ways to collect functions automatically? Is there a dedicated module for reflection in `python`?

Reply:
@skim0119 & @armantekinalp

Reply:
I don't think we need to modify anything for now, and it is working great! I'll open another issue(or discussion) and document this part with some suggestions since this conversation here will become difficult to track.

Reply:
I agree with @skim0119 . Current implementation works nicely, we need to add some more documentation for developers.

Reply:
Okay, but I agree with you that if there is an easier method, we should approach that instead. As @skim0119 suggested, lets track it in a separate issue.